libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java

```diff
@@ -153,20 +147,21 @@ private void initialize() {
       addMilestone(new VoiceInstructionMilestone.Builder().setIdentifier(VOICE_INSTRUCTION_MILESTONE_ID).build());
       addMilestone(new BannerInstructionMilestone.Builder().setIdentifier(BANNER_INSTRUCTION_MILESTONE_ID).build());
     }
-    if (options.snapToRoute()) {
-      snapEngine = new SnapToRoute();
-    }
-    if (options.enableOffRouteDetection()) {
-      offRouteEngine = new OffRouteDetector();
-    }
-    if (options().enableFasterRouteDetection()) {
-      fasterRouteEngine = new FasterRouteDetector();
+  }
+
+  private void initializeContext(Context context) {
+    if (applicationContext == null) {
```

Comment:
This is always `true` when initializing the context. We should remove this check.

Reply:
good catch 👍